toptal/picasso
# feat(PasswordInput): add new component

[FX-2482]

### Description

- this PR is about adding a new component called PasswordInput. It will have a toggle button to show and hide the password.

### How to test
You can see and test the new component here: https://picasso.toptal.net/fx-2482-create-passwordinput-with-hide-eye/?path=/story/forms-passwordinput--passwordinput

### Screenshots
<img width="346" alt="Screen Shot 2022-02-21 at 13 53 44" src="https://user-images.githubusercontent.com/7733167/154941411-c39ec308-dc6f-4c3c-8982-cd2a1f794351.png">
<img width="346" alt="Screen Shot 2022-02-21 at 13 53 47" src="https://user-images.githubusercontent.com/7733167/154941417-ee6fd2f2-6928-4a30-9803-8233b0a2a9ec.png">

### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-2482]: https://toptal-core.atlassian.net/browse/FX-2482?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
